io-sim/src/Control/Monad/IOSim/Internal.hs

```diff
@@ -779,6 +858,7 @@ reschedule !simstate@SimState{ threads, timers, curTime = time } =
     -- Note that 'threadDelay' is not handled via STM style wakeup, but rather
     -- it's handled directly above with 'wakeupThreadDelay' and 'unblockThreads'
     timeoutSTMAction (TimerThreadDelay _)     = return ()
+    timeoutSTMAction (TimerTimeout _ _ _)     = return ()
```

Comment:
I think we can safely use `{}`, if these constructors get a new field they will not be modified here.
```suggestion
    timeoutSTMAction TimerThreadDelay {}     = return ()
    timeoutSTMAction TimerTimeout {}         = return ()
```